x/ibc/03-connection/keeper/verify.go

```diff
@@ -31,9 +31,15 @@ func (k Keeper) VerifyClientConsensusState(
 		return sdkerrors.Wrapf(clienttypes.ErrConsensusStateNotFound, "clientID: %s with height: %d", clientID, height)
 	}
 
-	return clientState.VerifyClientConsensusState(
+	clientState, err = clientState.VerifyClientConsensusState(
 		k.aminoCdc, targetConsState.GetRoot(), height, connection.GetCounterparty().GetClientID(), consensusHeight, connection.GetCounterparty().GetPrefix(), proof, consensusState,
 	)
+
+	if err == nil {
+		k.clientKeeper.SetClientState(ctx, clientState)
+	}
+
+	return err
```

Comment:
the inverse is more readable:
```suggestion
	if err != nil {
		return err
	}
	
	k.clientKeeper.SetClientState(ctx, clientState)
	return nil
```